tests/test_json.py

```diff
@@ -0,0 +1,190 @@
+from fastavro import json_writer, json_reader
+from fastavro.six import StringIO
+
+import pytest
+
+pytestmark = pytest.mark.usefixtures("clean_readers_writers_and_schemas")
+
+
+def roundtrip(schema, records):
+    new_file = StringIO()
+    json_writer.writer(new_file, schema, records)
+    new_file.seek(0)
+
+    new_records = list(json_reader.reader(new_file, schema))
+    return new_records
+
+
+def test_json():
+    schema = {
+        "type": "record",
+        "name": "Test",
+        "namespace": "test",
+        "fields": [{
+            "name": "null",
+            "type": "null",
+        }, {
+            "name": "boolean",
+            "type": "boolean",
+        }, {
+            "name": "string",
+            "type": "string",
+        }, {
+            "name": "bytes",
+            "type": "bytes",
+        }, {
+            "name": "int",
+            "type": "int",
+        }, {
+            "name": "long",
+            "type": "long",
+        }, {
+            "name": "float",
+            "type": "float",
+        }, {
+            "name": "double",
+            "type": "double",
+        }, {
+            "name": "fixed",
+            "type": {
+                "type": "fixed",
+                "name": "fixed_field",
+                "size": 5,
+            },
+        }, {
+            "name": "union",
+            "type": [
+                'null',
+                'int',
```

Comment:
I would add more types here and test cases where union type is enum, fixed, float, double, etc. 
As many avro types have more than 1 correspondent in python, the complete implementation should handle more corner cases, like defining that the library should accept `decimal` as python correspondent for `double`, for instance. 

What would you do, for example, if you had 2 types in this union: string and bytes ? Both correspond to python strings.

I think basic cases like this are already worth merging, but they aren't the full thing. It would be possible to support something even more complete than current python avro implementation.

Reply:
I agree. I've been thinking about this recently and it would be really nice to have some helper (or maybe a standalone library) that given a schema it would produce 1) a valid datum for that schema using dummy values and 2) be able to create all the different types of valid datum (so in the case of unions, create multiple dictionaries where each is a different union case).

Without that existing, I think as you mentioned there should be some more common test cases. I just haven't added those yet as I wanted to just see if things were on the right track first. I'll add a comment to my top comment to say what needs to be done.

Reply:
Maybe I could write this library, let's just keep discussing first to reach common agreement. I've already spent a lot of time handling many of these complications.